### PR TITLE
feat: New integration tests

### PR DESCRIPTION
**Pull request by akim-bow**

### CODE REVIEW

File: integration-tests/src/deal-matcher.test.ts [hunk 1]
```diff
@@ -250,45 +154,62 @@ describe("#getMatchedOffersByDealId", () => {
...
-        matchedOffersOut.offers,
-        matchedOffersOut.computeUnitsPerOffers,
-      );
+      console.log(matchedOffersOut);
+
+      // const dealId = "0xd79df1927718b3212fa6e126ec4ad2b3ee1263d9"
+      console.log("---- Deal Matching ----");
 
-      const matchDealTxReceipt = await matchDealTx.wait(DEFAULT_CONFIRMATIONS);
+      // expect(matchedOffersOut.offers.length).toBe(1); // At least with one previously created offer it matched.
```
Comment (AlcibiadesCleinias) [inline integration-tests/src/deal-matcher.test.ts:185]: ph, I forgot when we will unlock this part?

File: integration-tests/src/offer.test.ts [hunk 2]
```diff
@@ -0,0 +1,343 @@
+import { beforeAll, describe, expect, test } from "vitest";
+import { registerMarketOffer } from "./helpers.js";
+import { getPeerFixture, getUnitIdFixture, randomCID } from "./fixtures.js";
+import { DEFAULT_CONFIRMATIONS } from "./constants.js";
+import { confirmEvents } from "./confirmations.js";
+import { ethers, JsonRpcProvider, JsonRpcSigner } from "ethers";
+import { type ContractsENV, DealClient } from "@fluencelabs/deal-ts-clients";
+import dns from "node:dns/promises";
+
+const ip = await dns.lookup("akim-dev.dev.fluence.dev");
```
Comment (AlcibiadesCleinias) [inline integration-tests/src/offer.test.ts:10]: what this IP is about?

**Review by AlcibiadesCleinias — CHANGES_REQUESTED** [review]

File: integration-tests/src/offer.test.ts (hunk 2 above)
Comment (akim-bow) [reply integration-tests/src/offer.test.ts:10]: Removed, thanks

File: integration-tests/src/deal-matcher.test.ts (hunk 1 above)
Comment (akim-bow) [reply integration-tests/src/deal-matcher.test.ts]: Later, need to interatively enable tests one by one

File: .github/workflows/artifacts.yml [hunk 3]
```diff
@@ -301,6 +301,85 @@ jobs:
...
+        working-directory: .github
+        env:
+          CHAIN_RPC_IMAGE: ${{ needs.chain-rpc.outputs.chain-rpc-image }}
+          CHAIN_DEPLOY_SCRIPT_IMAGE: ${{ needs.chain-rpc.outputs.chain-deploy-script-image }}
+          SUBGRAPH_DEPLOY_SCRIPT_IMAGE: ${{ needs.subgraph.outputs.subgraph-deploy-script-image }}
+
+      - name: Run integration tests
+        run: npm run test deal.test
+        working-directory: integration-tests
+
```
Comment (nahsi) [inline .github/workflows/artifacts.yml:382]:
```suggestion
```
It should go to test.yml

File: integration-tests/package.json [hunk 4]
```diff
@@ -15,7 +15,7 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
-    "@fluencelabs/deal-ts-clients": "file:../ts-client",
+    "@fluencelabs/deal-ts-clients": "file:fluencelabs-deal-ts-clients-0.7.0.tgz",
```
Comment (nahsi) [inline integration-tests/package.json:18]:
```suggestion
    "@fluencelabs/deal-ts-clients": "file:../ts-client",
```
Or else we need to build this archive manually on every change

File: integration-tests/src/offer.test.ts (hunk 2 above)
Comment (nahsi) [inline integration-tests/src/offer.test.ts:10]: Isn't it a temporary host that will go down any minute? :) Why it has to connect there?

**Review by nahsi — CHANGES_REQUESTED** [review]

File: integration-tests/src/offer.test.ts (hunk 2 above)
Comment (akim-bow) [reply integration-tests/src/offer.test.ts:10]: Thanks! I forgot to push these changes which remove that.

File: .github/workflows/artifacts.yml (hunk 3 above)
Comment (akim-bow) [reply .github/workflows/artifacts.yml:382]: Thanks! Moved

**Pull request closed without merging**